src/term/mod.rs

```diff
@@ -460,7 +463,7 @@ impl Term {
 
         while {
```

Comment:
I opted to keep the `!self.semantic_escape_chars.contains(...)` in the condition because there are two cases where we want to stop iteration:
 - There is nothing left to iterate over in that direction
 - Cell contains an escape character